svm-builds/build.rs

```diff
@@ -12,6 +14,15 @@ use svm::Releases;
 /// - "windows-amd64"
 pub const SVM_TARGET_PLATFORM: &str = "SVM_TARGET_PLATFORM";
 
+/// The path to the releases JSON file, that was pre-fetched manually. If this
+/// variable is defined, svm-builds won't attempt to deduce anything about the
+/// platfrom or perform network calls, and will instead treat the file as the
+/// source of truth.
+///
+/// Must follow the same format as the upstream `list.json`, e.g.
+/// [this one](https://raw.githubusercontent.com/nikitastupin/solc/af2fce8988e41753ab4f726e0273ea8244de5dba/linux/aarch64/list.json)
+pub const SVM_RELEASES_LIST_JSON: &str = "SVM_RELEASES_LIST_JSON";
```

Comment:
this is also nice to have,
I think what would be even more useful is having copies of all the lists in the repo itself which is used as fallback.

I guess we can write a test which serves as a helper that downloads all the lists

Reply:
I'd say that bundling the lists as a 'fallback' is a worse option as it make deterministic builds even worse. Now you might end up with different results depending of if your internet connection is flaky or not.

While I think that the solution proposed here is reasonable for the consumer as it gives them the freedom of deciding if they want to bundle versions (and if so—which versions), I can clearly see this being a problem for the concept of the version manager as a whole.

I don't think the other version managers I've stumbled on (like nvm and rvm) bundle the releases into themselves, but to make that route deterministic and reasonably reproducible, I'd suggest the following:

Instead of fetching the lists in the build.rs script, you *always* bundle them. That makes it a very reliable source because every release has a known list of versions. You can automate that so it's not an unreasonable burden, by running a github action daily that fetches the lists, checks if there are any diffs, and cutting off a new minor release if there are. It allows me to take my cargo.lock, move the repo to a new machine and still see the exact same versions that I expected to see. To update the list, you'd do the cargo update as usual, making sure you never pin the minor version.

What do you think?

Reply:
I think this is a reasonable fix. Thanks for contributing :)